### PR TITLE
[FW] Client-side form validation

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

This adds client-side validation and error handling for FW rule editing. 

### Validation
The form inside the drawer isn't validated by the API until the entire ruleset is submitted. Since the API has extensive validation and is the source of truth, the client doesn't try to validate 100% in sync with the API. A few examples:

- The client validates that `ports` contains only numbers, hyphens, and commas, but it doesn't check that the ranges make sense (e.g. `100-99` passes client but not server).
- The client validates IP address/range formatting, but not the feasibility of a range (e.g. `1.2.3.4/16` passes client but not server).

### Error handling
The API gives helpful error messages with great "field" granularity, e.g. `{ "field": "rules.inbound[0].addresses.ipv4[0]" }`. This PR provides a mechanism to parse these errors and map them to our internal structure of the ruleset.

**NOTE:** Better UX for displaying field errors is coming in a future PR. This will probably include NOT showing the actual error messages in the table, but rather an tooltip icon or something similar.

### CODE REVIEW

File: packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRuleDrawer.test.tsx [hunk 1]
```diff
@@ -34,38 +41,123 @@ describe('AddRuleDrawer', () => {
...
+      );
+    });
+    it('validates ports', () => {
+      expect(validateForm('ICMP', '80')).toHaveProperty(
+        'ports',
+        'Ports are not allowed for ICMP protocols.'
+      );
+      expect(validateForm('TCP')).toHaveProperty('ports');
+      expect(validateForm('UDP')).toHaveProperty('ports');
+      expect(validateForm('TCP', 'invalid-port')).toHaveProperty('ports');
```
Comment (Jskobos) [inline packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRuleDrawer.test.tsx:89]: I agree with not trying to replicate back-end validation here, but maybe the regex that's checking ports could be a bit stricter. As is `,,,,,` passes client validation and gets send to the API. Maybe something like `/([0-9\-]+,?)+/` would work

Comment (johnwcallahan) [reply packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRuleDrawer.test.tsx:89]: Good call; addressed.

**Comment by johnwcallahan** [comment]
@Jskobos great catch with that bug. I pushed up a fix. 

As for Yup validation, I did explore that but it proved to be too complicated for this PR. I'll open a ticket so we can explore later. (M3-4147)

File: packages/manager/src/features/Firewalls/FirewallDetail/Rules/firewallRuleEditor.ts [hunk 2]
```diff
@@ -164,23 +165,26 @@ export const editorStateToRules = (
...
-  rules: ExtendedFirewallRule[]
-): FirewallRuleType[] =>
-  rules.map(thisRule => ({ ...thisRule, errors: undefined }));
-
-export const removeStatus = (
-  rules: ExtendedFirewallRule[]
-): FirewallRuleType[] =>
-  rules.map(thisRule => ({ ...thisRule, status: undefined }));
+// Remove fields we use internally.
+export const stripExtendedFields = (
```
Comment (Jskobos) [inline packages/manager/src/features/Firewalls/FirewallDetail/Rules/firewallRuleEditor.ts:169]: Ramda's `pick()` might be nice here

Comment (johnwcallahan) [reply packages/manager/src/features/Firewalls/FirewallDetail/Rules/firewallRuleEditor.ts:169]: Good idea. I used `pick()`'s more pessimistic cousin, `omit()`.

**Pull request merged**